crates/sui-graphql-rpc/src/types/move_type.rs

```diff
@@ -173,7 +173,7 @@ impl TryFrom<TypeTag> for MoveTypeSignature {
             T::Vector(v) => Self::Vector(Box::new(Self::try_from(*v)?)),
 
             T::Struct(s) => Self::Struct {
-                package: format!("0x{}", s.address.to_canonical_string()),
+                package: s.address.to_canonical_display(true).to_string(),
```

Comment:
Nit: If we're really just creating a standalone string, then it's fine to use `to_canonical_string(x)` rather than `to_canonical_display(x).to_string()`.  The pattern to avoid is to create the string to put write back into a format string.

Also nit: Can you add a comment on the bool parameter to explain what it's for?

```suggestion
                package: s.address.to_canonical_string(/* with_prefix */ true),
```

The other option is to bind the flag to a variable first:

```rust
let with_prefix = true;
let _ = s.address.to_canonical_string(with_prefix);
```

We use both styles.

Reply:
Do we have thoughts on if we maybe use an enum vs a bool for clarity For this parameter?

Reply:
Not strong thoughts -- I like the type safety and intentionality of it, but in practice, it's a little bit cumbersome to use.  There's also the slightly left-of-field option of defining:

```rust
struct WithPrefix(bool);

fn to_canonical_display(&self, WithPrefix(with_prefix): WithPrefix) -> impl Display + '_ { /* ... */ }

s.to_canonical_display(WithPrefix(true));
```

It's a kind of middle ground, but I hesitate to wholeheartedly endorse it because it's not something we do across the codebase.

Reply:
oh see i would have done something like:
```
enum Prefix {
    Enabled,
    Disabled,
}
```
Bikeshedding not-with-standing. But i'm fine with keeping as-is in this PR,